OpenRobertaServer/src/test/java/de/fhg/iais/roberta/javaServer/basics/restInterface/GroupRestTest.java

```diff
@@ -0,0 +1,620 @@
+        User pid = userDao.get(pidId);
+
+        //changing existing groupMember
+        restGroups(
+            this.sPid,
+            "{ 'cmd':'updateMemberAccount';'groupName':'PidsGroup';'currentGroupMemberAccount':'PidsGroup:Member1';'newGroupMemberAccount':'UpdatedMember1' }",
+            "ok",
+            Key.SERVER_SUCCESS);
+
+        //check if groupInfo is now changed
+        restGroups(this.sPid, "{'cmd':'getUserGroup';'groupName':'PidsGroup'}", "ok", Key.GROUP_GET_ONE_SUCCESS);
+
+        JSONObject groupInfo = new JSONObject((String) this.response.getEntity()).getJSONObject("userGroup");
+        JSONArray members = groupInfo.getJSONArray("members");
+        List<JSONObject> member = IntStream.range(0, members.length()).mapToObj(members::getJSONObject).collect(Collectors.toList());
+
+        //check if Member1 is now replaced with UpdatedMember1
+        Assert.assertFalse(member.stream().anyMatch(p -> p.getString("account").equals("PidsGroup:Member1")));
+        Assert.assertTrue(member.stream().anyMatch(p -> p.getString("account").equals("PidsGroup:UpdatedMember1")));
+
+        //trying to create an already existing user --> should fail
+        restGroups(
+            this.sPid,
+            "{ 'cmd':'updateMemberAccount';'groupName':'PidsGroup';'currentGroupMemberAccount':'';'newGroupMemberAccount':'UpdatedMember1' }",
+            "error",
+            Key.GROUP_MEMBER_ERROR_ALREADY_EXISTS);
+
+        //giving no GroupMember should create a new one
+        restGroups(
+            this.sPid,
+            "{ 'cmd':'updateMemberAccount';'groupName':'PidsGroup';'currentGroupMemberAccount':'';'newGroupMemberAccount':'UpdatedMember2' }",
+            "ok",
+            Key.GROUP_ADD_MEMBER_SUCCESS);
+        //check if new member was created: this time with DAO
+
+        UserGroupDao groupDao = new UserGroupDao(newDbSession());
+        List<User> groupUsers = new ArrayList<>(groupDao.load("PidsGroup", pid).getMembers());
+        Assert.assertTrue(groupUsers.stream().anyMatch(p -> p.getAccount().equals("PidsGroup:UpdatedMember2")));
+
+        //trying to change a user without giving a new Account
+        restGroups(
+            this.sPid,
+            "{ 'cmd':'updateMemberAccount';'groupName':'PidsGroup';'currentGroupMemberAccount':'PidsGroup:UpdatedMember2';'newGroupMemberAccount':'' }",
+            "ok",
+            Key.SERVER_SUCCESS);
+        //check if groupInfo is now changed
+        restGroups(this.sPid, "{'cmd':'getUserGroup';'groupName':'PidsGroup'}", "ok", Key.GROUP_GET_ONE_SUCCESS);
+        JSONObject groupInfo2 = new JSONObject((String) this.response.getEntity()).getJSONObject("userGroup");
+        JSONArray members2 = groupInfo2.getJSONArray("members");
+        List<JSONObject> member2 = IntStream.range(0, members2.length()).mapToObj(members2::getJSONObject).collect(Collectors.toList());
+
+        //this may be a mistake the name of the member is now only the groupName and :
```

Comment:
this needs to be discussed:
 this happens when you update a member account but give it an empty string as an argument, this is either OK because in practice the front-end wont let you put in empty strings or the REST-Call should fail because this shouldn't happen.